solutions/ac-GCDOnBlackboard.mdx

```diff
@@ -114,4 +114,64 @@ print(ans)
 ```
 
 </PySection>
+
+	
+<JavaSection>
+	
+```java
+import java.util.*;
+import java.io.*;
+
+public class GCDBoard {
+	static final int MAX_N = (int) 1e5 + 5;
+
+	public static void main(String[] args) throws IOException {
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		int N = Integer.parseInt(br.readLine());
+		StringTokenizer st = new StringTokenizer(br.readLine());
+
+		int[] arr = new int[MAX_N];
+		for (int x = 1; x <= N; x++) {
+			arr[x] = Integer.parseInt(st.nextToken());
+		}
+		// Prefix will find the GCD from the start
+		int[] gcdPrefix = new int[MAX_N];
+
+		// Suffix will find the GCD from the end
+		int[] suffix = new int[MAX_N];
+
+		gcdPrefix[0] = 0;
+		suffix[N + 1] = 0;
```

Comment:
idk how java works but isn't the array already filled with 0s?

Reply:
yes it is, so this wouldnt be needed